src/cursor/cur_hs.c

```diff
@@ -199,15 +206,324 @@ __curhs_set_key(WT_CURSOR *cursor, ...)
 {
     WT_CURSOR *file_cursor;
     WT_CURSOR_HS *hs_cursor;
+    WT_ITEM *ds_key;
+    WT_SESSION_IMPL *session;
+    wt_timestamp_t start_ts;
+    uint64_t counter;
+    uint32_t arg_count;
     va_list ap;
 
     hs_cursor = (WT_CURSOR_HS *)cursor;
     file_cursor = hs_cursor->file_cursor;
+    session = CUR2S(cursor);
+    start_ts = WT_TS_NONE;
+    counter = 0;
 
     va_start(ap, cursor);
-    file_cursor->set_key(file_cursor, va_arg(ap, uint32_t), va_arg(ap, WT_ITEM *),
-      va_arg(ap, wt_timestamp_t), va_arg(ap, uint64_t));
+    arg_count = va_arg(ap, uint32_t);
+
+    WT_ASSERT(session, arg_count >= 1 && arg_count <= 4);
+
+    hs_cursor->btree_id = va_arg(ap, uint32_t);
+    F_SET(hs_cursor, WT_HSC_BTREE_ID_SET);
+    if (arg_count > 1) {
```

Comment:
I think use of switch statement will result in a relatively neat implementation.

Reply:
I don't think so. Here we are doing work progressively. If we use switch, we need to reverse the switch order, like:

switch () {
case 4:
....
case 3:
...
case 2:
...
}

Reply:
I'd prefer reverse order compared to the current long chain of if-else statements.

Reply:
Actually you can't reverse the order as va_arg cannot get variables in reverse order.